### DISCUSSION TITLE
👀 5.0的性能提升很大吗

### DISCUSSION THREAD

**Discussion by ShangChien** [antvis/G6#4602] [Q&A] [answered]
错过了直播，看了下官网首页的[demo](https://g6.antv.antgroup.com/g6v5)，感觉还不如之前的[案例](https://g6.antv.antgroup.com/zh/examples/performance/perf/#eva)流畅：

- 白屏时间较长，
- 拖动卡顿，甚至页面卡死。

还是说5.0的提升，不是着重在性能方面？

**Comment by pomelo-nwu** [comment]
@ShangChien Alpha 版本嘛，底层用的是G：https://g.antv.antgroup.com/examples/perf/basic/#nodes-5w  做了重大升级。到上层G6这一层，还需要时间打磨优化。正式发布的 5.0 最重要的提升就是性能方面，这点是确定的

**Reply by ShangChien** [reply]
回看了直播，可控的细节增加了好多，值得期待。

**Reply by pomelo-nwu** [reply]
5.0 工程架构的升级也是非常重要的，比如我们上层的产品 G6VP，就需要依赖一些机制去做更多的事情：https://github.com/antvis/G6VP

**Comment by Yanyan-Wang** [comment, marked as answer]
我们测试的时候有发现开了 sketch 、keynote 这种动不动就吃几十个 G 性能的 app 的时候，运行 demo 的交互和动画都挺卡的，一关掉就会流长很多。但是理论上性能应该是比 4.x 要好很多。你贴的这个 4.x 的 demo 拖拽平移起来还可以，是因为过程中把除了节点的 keyshape 以外的图形都隐藏了。而 5.x 的这个 demo 没有做这个优化，保留了当前可以看到的所有图形。

当然 5.x 的性能还有优化的空间，继续努力💪

**Comment by Yanyan-Wang** [comment]
除了渲染和交互性能以外，还有布局性能的提升，有了 rust + WASM 并行计算。这里直播回看地址：https://www.bilibili.com/video/BV16u4y1d7kv/?spm_id_from=333.788&vd_source=49df598f9a516023271b790c7c5d81d8

**Reply by rmd1710714107** [reply]
关于rust和wasm的使用有案例吗？我按着官方文档方法使用会报错
